internal/aws/proxy/conn.go

```diff
@@ -103,15 +115,19 @@ func getAWSConfigSession(c *Config, logger *zap.Logger) (*aws.Config, *session.S
 		awsRegion, err = getRegionFromECSMetadata()
 		if err != nil {
 			logger.Debug("Unable to fetch region from ECS metadata", zap.Error(err))
-			var sess *session.Session
-			sess, err = session.NewSession()
-			if err == nil {
-				awsRegion, err = getEC2Region(sess)
+			config, innerError := awsConfig.LoadDefaultConfig(context.Background())
+			config.Retryer = func() awsSDKV2.Retryer {
```

Comment:
nit - No point in setting `config.Retryer` if there was an `innerError`. You could move the assignment after checking for an error.

Reply:
innerError means it could not set up config. I was unable to set that var to err since it would have been a shadow var. Giving us the err value for the next line check. The retryer is for when we try to reach out to imds with the clinet.